### PR TITLE
new nuopc/cmeps single/column single/point functionality

### PR DESCRIPTION
**Pull request by mvertens**
Implemented new nuopc/cmeps single column functionality.

1. In `config/cesm/config_files.xml` - single point domains are only used for mct/cpl7. For cmeps single point meshes are now generated on the fly and component domains files are no longer needed.
2. env_run.xml variables PTS_LAT, PTS_LON and PTS_DOMAINFILE are used to determine if there is a single point or single column run. 
  - If PTS_LAT and PTS_LON are not -999 and PTS_DOMAINFILE is UNSET, then you have a single point run and the exact values of PTS_LAT and PTS_LON are used. 
  - If PTS_LAT and PTS_LON are not -999 and PTS_DOMAINFILE is not UNSET, then then the cmeps driver will recognize the nearest neighbor values of PTS_LAT and PTS_LON in PTS_DOMAINFILE as the single column lat and lon to use.

This PR only effects nuopc/cmeps configurations and also needs an accompanying PR to CMEPS, CDEPS, CLM, CICE5 and CAM. A full documentation of the testing is done in the CMEPS PR.

Test suite:
Test baseline:
Test namelist changes: 
Test status: bit for bit and roundoff 

Fixes: None
User interface changes?: 
Update gh-pages html (Y/N)?: N

Code review:

### CODE REVIEW

**Review by jedwards4b — APPROVED** [review]

**Comment by jedwards4b** [comment]
@mvertens I'll try to get the github actions working again - this may involve several commits to your branch.

File: config/cesm/config_grids.xml [hunk 1]
```diff
@@ -1920,9 +1862,112 @@
...
   <!-- ======================================================== -->
 
+  <domains driver="mct">
+    <domain name="01col">
+      <nx>1</nx> <ny>1</ny>
+      <file>domain.ocn.01col.ArcticOcean.20150824.nc</file>
+      <file>domain.ocn.01col.ArcticOcean.20150824.nc</file>
+      <desc>01col is a single-column grid for datm and POP:</desc>
+    </domain>
+    <domain name="CLM_USRDAT">
```
Comment (ekluzek) [inline config/cesm/config_grids.xml:1875]: Because of #3576 I recommend you remove this setting for CLM_USRDAT for a specific domain file.

Comment (mvertens) [reply config/cesm/config_grids.xml]: Done

Comment (ekluzek) [reply config/cesm/config_grids.xml]: Perfect. Thank you!

File: .github/workflows/srt.yml [hunk 2]
```diff
@@ -39,7 +39,7 @@ jobs:
 
       - id: load-env
         run: |
-          sudo apt-get update
+          #sudo apt-get update
```
Comment (ekluzek) [inline .github/workflows/srt.yml:42]: Just making sure this is on purpose.

**Review by ekluzek — APPROVED** [review]
I ask one question about a change. But, everything else looks good to me.

File: .github/workflows/srt.yml (hunk 2 above)
Comment (jedwards4b) [reply .github/workflows/srt.yml:42]: @ekluzek Yes it's intentional we can remove the commented line if you prefer.

Comment (ekluzek) [reply .github/workflows/srt.yml:42]: I think removing the line is a better practice when we have the history in version control. The only reason to comment it out is if there are some conditions where you'd add it back in again. But, then it should have a comment about why you'd do that.

**Comment by rljacob** [comment]
For reasons we are still trying to figure out, this broke an E3SM case at create_newcase stage:


```
./create_newcase --case testGcase2 --res TL319_EC30to60E2r2 --compset GMPAS-JRA1p4
Compset longname is 2000_DATM%JRA-1p4-2018_SLND_MPASSI_MPASO%DATMFORCED_DROF%JRA-1p4-2018_SGLC_SWAV
Compset specification file is /gpfs/fs1/home/jacob/E3SM3/components/mpas-ocean/cime_config/config_compsets.xml
Automatically adding SIAC to compset
Automatically adding SESP to compset
ERROR: Expected one child, found None with name 'domains' and attribs '{'driver': 'mct'}' in file /gpfs/fs1/home/jacob/E3SM3/cime_config/config_grids.xml
Exit 1
chrlogin2.lcrc.anl.gov[187]: ./create_newcase --debug --case testGcase2 --res TL319_EC30to60E2r2 --compset GMPAS-JRA1p4
RUN: /usr/bin/xmllint --xinclude --noout --schema /gpfs/fs1/home/jacob/E3SM3/cime/config/xml_schemas/entry_id.xsd /gpfs/fs1/home/jacob/E3SM3/cime/config/e3sm/config_files.xml
FROM: /gpfs/fs1/home/jacob/E3SM3/cime/scripts
  errput: /gpfs/fs1/home/jacob/E3SM3/cime/config/e3sm/config_files.xml validates

RUN: /usr/bin/xmllint --xinclude --noout --schema /gpfs/fs1/home/jacob/E3SM3/cime/config/xml_schemas/config_compsets.xsd /gpfs/fs1/home/jacob/E3SM3/cime/src/drivers/mct/cime_config/config_compsets.xml
FROM: /gpfs/fs1/home/jacob/E3SM3/cime/scripts
  errput: /gpfs/fs1/home/jacob/E3SM3/cime/src/drivers/mct/cime_config/config_compsets.xml validates

Compset longname is 2000_DATM%JRA-1p4-2018_SLND_MPASSI_MPASO%DATMFORCED_DROF%JRA-1p4-2018_SGLC_SWAV
Compset specification file is /gpfs/fs1/home/jacob/E3SM3/components/mpas-ocean/cime_config/config_compsets.xml
RUN: /usr/bin/xmllint --xinclude --noout --schema /gpfs/fs1/home/jacob/E3SM3/cime/config/xml_schemas/entry_id.xsd /gpfs/fs1/home/jacob/E3SM3/cime/src/drivers/mct/cime_config/config_component.xml
FROM: /gpfs/fs1/home/jacob/E3SM3/cime/scripts
  errput: /gpfs/fs1/home/jacob/E3SM3/cime/src/drivers/mct/cime_config/config_component.xml validates

Automatically adding SIAC to compset
Automatically adding SESP to compset
> /gpfs/fs1/home/jacob/E3SM3/cime/scripts/lib/CIME/utils.py(136)expect()
-> msg = error_prefix + " " + error_msg
(Pdb) 

```

**Comment by jedwards4b** [comment]
@rljacob I can try to work with someone on your team to debug this - anyone available?

**Pull request merged**